File: backend/src/main/java/com/project2/domain/post/dto/PostFormResponseDTO.java
package com.project2.domain.post.dto;

import java.util.List;

import com.project2.domain.post.entity.Post;
import com.project2.domain.post.entity.PostImage;

import lombok.Getter;

@Getter
public class PostFormResponseDTO {

	private final String title;

	private final String content;

	private final Long placeId;

	private final Double latitude;

	private final Double longitude;

	private final String placeName;

	private final String category;

	private final String region;

	private final List<String> images;

	public PostFormResponseDTO(Post post) {
		this.title = post.getTitle();
		this.content = post.getContent();
		this.placeId = post.getPlace().getId();
		this.latitude = post.getPlace().getLatitude();
		this.longitude = post.getPlace().getLongitude();
		this.placeName = post.getPlace().getName();
		this.category = post.getPlace().getCategory().getKrCategory();
		this.region = post.getPlace().getRegion().getKrRegion();
		this.images = post.getImages().stream().map(PostImage::getImageUrl).sorted().toList();
	}
}

File: backend/src/main/kotlin/com/project2/domain/post/dto/PostFormResponseDTO.kt
package com.project2.domain.post.dto

import com.project2.domain.post.entity.Post

class PostFormResponseDTO(
        post: Post
) {
    val title: String = post.title
    val content: String = post.content
    val placeId: Long = post.place.id!!
    val latitude: Double = post.place.latitude
    val longitude: Double = post.place.longitude
    val placeName: String = post.place.name
    val category: String = post.place.category.krCategory
    val region: String = post.place.region.krRegion
    val images: List<String> = post.images.map { it.imageUrl }.sorted()
}